jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/udp/UdpHexDecoder.java

```diff
@@ -0,0 +1,245 @@
+        int senderPort = packet.getPort();
+        logger.debug("Packet received from {}:{}", senderIp, senderPort);
+
+        // Create OdeMsgPayload and OdeLogMetadata objects and populate them
+        OdeAsn1Payload ssmPayload = getPayloadHexString(packet, UperUtil.SupportedMessageTypes.SSM);
+        if (ssmPayload == null)
+            return null;
+        OdeSsmMetadata ssmMetadata = new OdeSsmMetadata(ssmPayload);
+
+        // Add header data for the decoding process
+        ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
+        String timestamp = utc.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
+        ssmMetadata.setOdeReceivedAt(timestamp);
+
+        ssmMetadata.setOriginIp(senderIp);
+        ssmMetadata.setSsmSource(SsmSource.RSU);
+        ssmMetadata.setRecordType(RecordType.ssmTx);
+        ssmMetadata.setRecordGeneratedBy(GeneratedBy.RSU);
+        ssmMetadata.setSecurityResultCode(SecurityResultCode.success);
+
+        return JsonUtils.toJson(new OdeAsn1Data(ssmMetadata, ssmPayload), false);
+    }
+
+    public static String buildJsonSrmFromPacket(DatagramPacket packet){
+        String senderIp = packet.getAddress().getHostAddress();
+        int senderPort = packet.getPort();
+        logger.debug("Packet received from {}:{}", senderIp, senderPort);
+
+        // Create OdeMsgPayload and OdeLogMetadata objects and populate them
+        OdeAsn1Payload srmPayload = getPayloadHexString(packet, UperUtil.SupportedMessageTypes.SRM);
+        if (srmPayload == null)
+            return null;
+        OdeSrmMetadata srmMetadata = new OdeSrmMetadata(srmPayload);
+
+        // Add header data for the decoding process
+        ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
+        String timestamp = utc.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
+        srmMetadata.setOdeReceivedAt(timestamp);
+
+        srmMetadata.setOriginIp(senderIp);
+        srmMetadata.setSrmSource(SrmSource.RSU);
+        srmMetadata.setRecordType(RecordType.srmTx);
+        srmMetadata.setRecordGeneratedBy(GeneratedBy.OBU);
+        srmMetadata.setSecurityResultCode(SecurityResultCode.success);
+
+        return JsonUtils.toJson(new OdeAsn1Data(srmMetadata, srmPayload), false);
+    }
+
+    public static String buildJsonPsmFromPacket(DatagramPacket packet){
+        String senderIp = packet.getAddress().getHostAddress();
+        int senderPort = packet.getPort();
+        logger.debug("Packet received from {}:{}", senderIp, senderPort);
+
+        // Create OdeMsgPayload and OdeLogMetadata objects and populate them
+        OdeAsn1Payload psmPayload = getPayloadHexString(packet, UperUtil.SupportedMessageTypes.PSM);
+        if (psmPayload == null)
+            return null;
+        OdePsmMetadata psmMetadata = new OdePsmMetadata(psmPayload);
+        // Add header data for the decoding process
+        ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
```

Comment:
We could probably use a method for creating this timestamp if duplicated in multiple places.

Reply:
Sounds good to me. I added a static method at the bottom that is shared between all the sources for this.